Update points

izeke opened the pull request:
Changes:
- Adds existing point display to submit page
- Adds color coded markers on submit page
- Changes home page markers to match submit page marker style
- Changes pothole_geojson view to take an active parameter
- Adds update_pothole callback which works similarly to submit_pothole but takes an existing pothole id
- Add confirm button that allows severity selection and calls update_pothole to all existing potholes on submit page
- Add mark fixed button that calls update_pothole to all confirmed and fixed potholes on submit page

frankjh3 commented on wheel_time_potholes/pothole_reporting/views.py:
```diff
@@ -73,6 +94,12 @@ def pothole_picture(request):
+        if active is not None and active.lower() != "false":
+            active = True
+        else:
+            active = False
```
Could be simplified to:
`active = active is not None and active.lower() != "false"`

frankjh3 commented on wheel_time_potholes/pothole_reporting/static/pothole_reporting/js/SubmissionMap.js:
```diff
@@ -47,12 +67,128 @@ function initMap() {
+    },
+    success: function (data) {
+      alert("Success");
+      placeMarker(latLng);
```
I'm not sure if this should call placeMarker, since there is already a marker corresponding to this pothole, unless that updates the marker to reflect its new status.

frankjh3 commented on wheel_time_potholes/pothole_reporting/static/pothole_reporting/js/SubmissionMap.js:
```diff
@@ -47,12 +67,128 @@ function initMap() {
+      xhr.setRequestHeader("X-CSRFToken", getCookie("csrftoken"));
+    },
+    success: function (data) {
+      alert("Success");
```
You could update the marker to reflect the added confirmation / fixed report in the browser with the following:
```
      if (fixed) {
        map.data.getFeatureById(activeFeature.getId())['j']['fixed_reports']++;
      } else {
        map.data.getFeatureById(activeFeature.getId())['j']['pothole_reports']++;
      }
```

izeke commented:
Looks like the fixed pothole not displaying as fixed issue was due to improper comparison of dates in the backend. We were using `datetime.now()` which returns local time, while we were inserting in utc. Fixed that by changing to `datetime.utcnow()`. I also replaced a few things with a method called `reloadGeojson()` which reloads geojson from the database. This is called when a marker is placed or updated, and changes are reflected immediately. For example, fixed or confirmed reports will increment by one, and newly submitted potholes will be hoverable immediately.
